components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -658,7 +658,11 @@ private String handleUserConsent(HttpServletRequest request, String consent, OAu
             return EndpointUtil.getErrorRedirectURL(oauthProblemException, oauth2Params);
         }
 
-        return oauthResponse.getBody() == null ? oauthResponse.getLocationUri() : oauthResponse.getBody();
+        //When responseType equal to "id_token" the resulting token is passed back as a query parameter
+        //According to the specification it should pass as URL Fragment
+        return oauthResponse.getBody() == null ?
```

Comment:
What happens if the callback URL contains a query param?
https://localhost/callback?param1=abc&id_token=<>